api/api_calls.py

```diff
@@ -671,7 +671,16 @@ def _sample_upload_generator(sequence_file_up):
         logging.debug("data:" + str(data_pkg))
         logging.debug("headers: " + str(headers_pkg))
 
-        response = self._session.post(url, data=data_pkg, headers=headers_pkg)
+        try:
+            response = self._session.post(url, data=data_pkg, headers=headers_pkg)
+        except ConnectionError as e:
+            # This could be anything from disconnection during post to IRIDA crashing
+            logging.error("ConnectionError occurred while transferring data: " + str(e))
+            raise exceptions.IridaConnectionError(e)
+        except Exception as e:
+            # Any other exception, like a library not handling the response properly could be caught here
+            logging.error("Exception occured while transferring data: " + str(e))
+            raise exceptions.IridaConnectionError(e)
 
```

Comment:
This will catch any weird errors that come from other libraries. This sometimes caused crashes.